File: 2016-reduction-operations-to-make-the-array-elements-equal/reduction-operations-to-make-the-array-elements-equal.java
class Solution {
    public int reductionOperations(int[] nums) {
        int ans = 0, max = 0;
        for(int num : nums) max = Math.max(max, num);
        int[] buckets = new int[max+1];
        for(int num : nums) buckets[num]++;
        int curr = buckets[max];
        for(int i=buckets.length-2; i>=0; i--) {
            if(buckets[i] > 0) {
                ans += curr;
                curr += buckets[i];
            }
        }
        return ans;
    }
}

File: 2016-reduction-operations-to-make-the-array-elements-equal/reduction-operations-to-make-the-array-elements-equal.py
class Solution:
    def reductionOperations(self, nums: List[int]) -> int:
        ans = 0
        maxi = max(nums)
        buckets = defaultdict(int)

        for num in nums:
            buckets[num] = buckets[num] + 1
        
        curr = buckets[maxi]

        for num in range(maxi-1, -1, -1):
            if buckets[num]:
                ans += curr
                curr += buckets[num]

        return ans